catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/plugins/FrameCenterUpdateField.java

```diff
@@ -0,0 +1,105 @@
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.alliance.libs.klv.AttributeNameConstants;
+import org.codice.alliance.libs.klv.GeometryOperator;
+import org.codice.alliance.libs.klv.GeometryUtility;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.io.WKTReader;
+import com.vividsolutions.jts.io.WKTWriter;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+/**
+ * Update the frame-center field based on the frame-center data in the children. The coordinates
+ * of the child frame-center linestrings are concatenated into a single linestring and saved to
+ * the parent.
+ * <p>
+ * This is not thread-safe.
+ */
+@NotThreadSafe
+public class FrameCenterUpdateField implements UpdateParent.UpdateField {
+
+    private final GeometryOperator geometryOperator;
+
+    private Geometry intermediateGeometry;
+
+    /**
+     * @param geometryOperator applied to the final linestring before it is saved to the parent
+     */
+    public FrameCenterUpdateField(GeometryOperator geometryOperator) {
+        this.geometryOperator = geometryOperator;
+    }
+
+    @Override
+    public String toString() {
+        return "FrameCenterUpdateField{" + "geometryOperator=" + geometryOperator
+                + ", intermediateGeometry=" + intermediateGeometry + '}';
+    }
+
+    @Override
+    public void updateField(Metacard parent, List<Metacard> children) {
+
+        WKTReader wktReader = new WKTReader();
+
+        List<String> childLocations = extractChildFrameCenters(children);
+
+        List<Geometry> geometries = new LinkedList<>();
+
+        if (intermediateGeometry != null) {
+            geometries.add(intermediateGeometry);
+        }
+
+        geometries.addAll(childLocations.stream()
+                .map(s -> GeometryUtility.wktToGeometry(s, wktReader))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList()));
+
+        List<Coordinate> coordinates = geometries.stream()
+                .flatMap(geometry -> Stream.of(geometry.getCoordinates()))
+                .collect(Collectors.toList());
+
+        intermediateGeometry =
+                new GeometryFactory().createLineString(coordinates.toArray(new Coordinate[coordinates.size()]));
+
+    }
+
+    @Override
+    public void end(Metacard parent) {
+        if (intermediateGeometry != null) {
+            setFrameCenter(parent, geometryOperator.apply(intermediateGeometry));
+        }
+    }
+
+    private void setFrameCenter(Metacard parentMetacard, Geometry geometry) {
+        WKTWriter wktWriter = new WKTWriter();
```

Comment:
I believe there are thread safety issues with WKTWriter.  In the GmdTransforer we handle that we a thread local object: https://github.com/codice/ddf/blob/08651d5cc59e559b940b4faaec6334657948c18c/catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java#L101